lib/matplotlib/layout_engine.py

```diff
@@ -101,6 +101,28 @@ def execute(self, fig):
         raise NotImplementedError
 
 
+class NullLayoutEngine(LayoutEngine):
+    """
+    A no-op LayoutEngine.
+
+    This is primarily to act as a place holder if we remove a layout engine.
```

Comment:
```suggestion
    This is acts as a place holder if we remove a layout engine.  However, different layout engines
    require different colorbar implementations and some are not compatible with ``subplots_adjust``, 
    so this Null engine retains the requirements of the last-used layout engine.  
```

Reply:
This is an improvement 👍 . Can you make it even more explicit: "different layout engines require different ..." / "some are not compatible". To my knowledge we have exactly two layout engines we are talking about.

Reply:
True, but the goal of this hook was to make it easy to add more / let external libraries provide their own so I think that leaving this general is the optimistic position.